codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/localservice/shapevisitor/conversionwriter/LocalServiceToDafnyConversionFunctionWriter.java

```diff
@@ -131,7 +132,7 @@ private void writeNonReferenceStructureShapeMemberConverter(
     if (context.model().expectShape(memberShape.getTarget()).hasTrait(ReferenceTrait.class)) {
       if (memberShape.isOptional()) {
         conversionWriter.write(
-            "((Option_Some($1L)) if (($2L is not None) and ($1L is not None)) else (Option_None())),",
+            "((Option_Some($1L)) if (($1L is not None) and ($2L is not None)) else (Option_None())),",
```

Comment:
The swap should be for the params:

```
"((Option_Some($2L)) if (($1L is not None) and ($2L is not None)) else (Option_None())),",
           dataSourceInsideConversionFunction + "." + CaseUtils.toSnakeCase(memberName),
            targetShape.accept(
                ShapeVisitorResolver.getToDafnyShapeVisitorForShape(
                    targetShape,
                    context,
                    dataSourceInsideConversionFunction + "." + CaseUtils.toSnakeCase(memberName),
                    conversionWriter,
                    "smithy_to_dafny")));
```
otherwise you might end up with NPE since with the new change there's no short circuiting.

Reply:
Now that I look at it again, I don't think what you're suggesting does anything.
This is *writing* code. Both params need to be written. There is no short circuiting going on. The if statement gets evaluated at runtime, not at codegen time.

Reply:
Yup, it has nothing to do with the generated code. What I meant was that it's easier to read the codegen placeholder code - _If 1 is not None then check for 2_ vs if _2 is not none then check for 1._

Reply:
Got it, done